### PR TITLE
Replace exclusionary words whitelist and blacklist in the places that…

### PR DESCRIPTION
**Pull request by aponb**
### Description
… won't impact backwards compatibility (#2178)

* Replace the exclusionary word whitelist with allowlist, and blacklist with denylist, in code comment and internal variable/method/class/package name.
 
### Issues Resolved
backport of #2178 for 1.x
 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Signed-off-by: Andreas <apre@gmx.at>

### CODE REVIEW

**Comment by tlfeng** [comment]
I think it can be merged after resolving the conflict in `plugins/discovery-azure-classic/build.gradle`.

**Comment by aponb** [comment]
> I think it can be merged after resolving the conflict in `plugins/discovery-azure-classic/build.gradle`.

anything I can do?

**Comment by tlfeng** [comment]
> anything I can do?

You will need to merge the upstream `1.x` branch to your local feature branch `1.x-backport2178`, and resolve the conflict.

or follow the instruct to use the web editor to resolve this conflict, and then add DCO sign-off to the commit.
image

Thanks!

**Comment by aponb** [comment]
> > anything I can do?
> 
> You will need to merge the upstream `1.x` branch to your local feature branch `1.x-backport2178`, and resolve the conflict.
> 
> or follow the instruct to use the web editor to resolve this conflict, and then add DCO sign-off to the commit. image
> 
> Thanks!

Oh I see. There were further commits. It's merged again now!

**Comment by tlfeng** [comment]
> Oh I see. There were further commits. It's merged again now!
Thank you! Yeah.. there are many new commits being merged recently.

**Comment by tlfeng** [comment]
There are 2 problems in the log 3133:
```
3: Task failed with an exception.
-----------
* Where:
Build file '/var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/search/qa/full-cluster-restart/build.gradle' line: 80

* What went wrong:
Execution failed for task ':qa:full-cluster-restart:v1.2.5#oldClusterTest'.
> `cluster{:qa:full-cluster-restart:v1.2.5}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':plugins:discovery-azure-classic:dependencyLicenses'.
> Missing SHA for jaxb-api-2.2.2.jar. Run "gradle updateSHAs" to create them
```
The first "failed for task ':qa:full-cluster-restart:v1.2.5#oldClusterTest'." is not likely caused by this PR, but the second one (`'javax.xml.bind:jaxb-api:2.3.1'`) is caused by a mistake in last commit.

File: plugins/discovery-azure-classic/build.gradle [hunk 1]
```diff
@@ -63,8 +63,8 @@ dependencies {
   api 'com.sun.xml.bind:jaxb-impl:2.2.3-1'
 
   // HACK: javax.xml.bind was removed from default modules in java 9, so we pull the api in here,
-  // and whitelist this hack in JarHell
-  api 'javax.xml.bind:jaxb-api:2.3.1'
+  // and allowlist this hack in JarHell
+  api 'javax.xml.bind:jaxb-api:2.2.2'
```
[the remark this answers is not in the document]
Comment (aponb) [reply plugins/discovery-azure-classic/build.gradle]: Respect. You have a lot of patience. Sorry for that, I missed that. It's fixed now.

[the remark this answers is not in the document]
Comment (tlfeng) [reply plugins/discovery-azure-classic/build.gradle]: You are very welcome! 😄 Glad you are so responsive.

**Review by tlfeng — APPROVED** [review]
I validated all the changes in the backport PR is the same with those that merged in `main` branch.

**Review by owaiskazi19 — APPROVED** [review]
Thanks for the changes!

**Pull request merged**